cmd/anonymizer/app/uiconv/extractor_test.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package uiconv
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/go-openapi/swag"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+type UITrace struct {
+	Data []model.Trace
+}
+
+func TestExtractor_TraceSuccess(t *testing.T) {
+	inputFile := "fixtures/trace_success.json"
+	outputFile := "fixtures/trace_success_ui_anonymized.json"
+	defer os.Remove(outputFile)
+
+	reader, err := NewReader(
+		inputFile,
+		zap.NewNop(),
+	)
+	require.NoError(t, err)
+
+	extractor, err := NewExtractor(
+		outputFile,
+		"2be38093ead7a083",
+		reader,
+		zap.NewNop(),
+	)
+	require.NoError(t, err)
+
+	err = extractor.Run()
+	require.NoError(t, err)
+
+	var trace UITrace
+	loadJSON(t, outputFile, &trace)
+
+	for i := range trace.Data {
+		for j := range trace.Data[i].Spans {
+			assert.Equal(t, "span.kind", trace.Data[i].Spans[j].Tags[0].Key)
+		}
+	}
+}
+
+func TestExtractor_TraceOutputFileError(t *testing.T) {
+	inputFile := "fixtures/trace_success.json"
+	outputFile := "fixtures/trace_success_ui_anonymized.json"
+	defer os.Remove(outputFile)
+
+	reader, err := NewReader(
+		inputFile,
+		zap.NewNop(),
+	)
+	require.NoError(t, err)
+
+	err = os.Chmod("fixtures", 0000)
+	require.NoError(t, err)
+	defer os.Chmod("fixtures", 0755)
+
+	_, err = NewExtractor(
+		outputFile,
+		"2be38093ead7a083",
+		reader,
+		zap.NewNop(),
+	)
+	require.Equal(t, "cannot create output file: open fixtures/trace_success_ui_anonymized.json: permission denied", err.Error())
```

Comment:
nit: use `require.Contains(t, "cannot create output file", err.Error())` instead of exact comparison, it makes tests more resilient to potential changes in the error messages in the stdlib, which are not the main subject of the test.

Reply:
That's a great suggestion. I didn't think about it at all. I've updated the PR accordingly.

Reply:
Ah I think I included msg from stdlib took, removing that.

Reply:
Updated the PR removing loglines from stdlib.